PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobLocalRunConfigurationProducer.java

```diff
@@ -0,0 +1,183 @@
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.util.PsiMethodUtil;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.microsoft.azure.hdinsight.spark.common.SparkBatchJobConfigurableModel;
+import com.microsoft.azure.hdinsight.spark.run.configuration.RemoteDebugRunConfiguration;
+import com.microsoft.azure.hdinsight.spark.run.configuration.RemoteDebugRunConfigurationType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.plugins.scala.runner.ScalaMainMethodUtil;
+import scala.Option;
+import scala.Tuple2;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.Optional;
+import java.util.Set;
+
+public class SparkBatchJobLocalRunConfigurationProducer extends JavaRunConfigurationProducerBase<RemoteDebugRunConfiguration> {
+    public SparkBatchJobLocalRunConfigurationProducer() {
+        super(RemoteDebugRunConfigurationType.getInstance());
+    }
+
+    @Override
+    protected boolean setupConfigurationFromContext(RemoteDebugRunConfiguration configuration, ConfigurationContext context, Ref<PsiElement> sourceElement) {
+        final Optional<Location> location = Optional.ofNullable(context.getLocation());
+
+        return location
+                .map(JavaExecutionUtil::stepIntoSingleClass)
+                .map(Location::getPsiElement)
+                .filter(PsiElement::isPhysical)
+                .flatMap(element -> {
+                    Optional<SimpleImmutableEntry<PsiElement, PsiClass>> mcPair = findMainMethod(element);
+
+                    if (mcPair.isPresent()) {
+                        return mcPair;
+                    } else {
+                        Optional<SimpleImmutableEntry<PsiElement, PsiClass>> ccPair = findMainClass(element);
+
+                        return ccPair.isPresent() ? ccPair : findScalaMainClass(element);
+                    }
+                })
+                .filter(mcPair -> {
+                    // To determine if the current context has Spark Context dependence
+                    DependenciesBuilder db = new ForwardDependenciesBuilder(
+                            context.getModule().getProject(), new AnalysisScope(mcPair.getKey().getContainingFile()));
+
+                    db.analyze();
+
+                    return Optional.ofNullable(db.getDependencies().get(mcPair.getKey().getContainingFile()))
+                            .map((Set<PsiFile> t) -> t.stream()
+                                    .map(PsiFile::getVirtualFile)
+                                    .map(VirtualFile::getName)
+                                    .anyMatch(className -> className.equals("SparkContext.class") || className.equals("JavaSparkContext.class")))
+                            .orElse(false);
+                })
+                .map(mcPair -> {
+                    sourceElement.set(mcPair.getKey());
+                    setupConfiguration(configuration, mcPair.getValue(), context);
```

Comment:
we will setup our run configuration if "SparkContext" or "JavaSparkContext" exists? What if customer just willing to run their own project without our configuraion?

Reply:
Fix it